rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/service/SpringBeanService.java

```diff
@@ -21,4 +26,22 @@ public SpringBeanService(GraphContext context)
     {
         return super.findAllByProperty(SpringBeanModel.SPRING_BEAN_NAME, name);
     }
+
+    /**
+     *
+     * @param application
+     * @return an unmodifiable list of SpringBeanModel entries for the given application
+     */
+    public List<SpringBeanModel> findAllByApplication(ProjectModel application)
+    {
+        List<SpringBeanModel> models = new ArrayList<>();
+        for (SpringBeanModel model : findAll())
+        {
+            if (application.equals(model.getApplication()))
+            {
+                models.add(model);
+            }
```

Comment:
Makes sense. I was not sure how to implement this properly. Anyway, we should probably review all other rules as well. For instance, CreateJPAReportRuleProvider gets and filters all the JPA entities...